components/support/base/src/main/java/mozilla/components/support/base/observer/Consumable.kt

```diff
@@ -52,6 +64,22 @@ class Consumable<T> private constructor(
     @Synchronized
     fun isConsumed() = value == null
 
+    /**
+     * Returns the value of this [Consumable] without consuming it.
+     */
+    @Synchronized
+    fun peek(): T? = value
+
+    /**
+     * Adds a listener to be invoked when this [Consumable] is consumed.
+     *
+     * @param listener the listener to add.
+     */
+    @Synchronized
+    fun onConsume(listener: ConsumableListener) {
+        listeners.add(listener)
+    }
+
```

Comment:
Should we add those two methods to the changelog? The Firefox for Fire TV team uses `Consumable` in their code. Maybe those could help them too.